sys/net/gnrc/network_layer/ndp/gnrc_ndp.c

```diff
@@ -193,6 +193,12 @@ void gnrc_ndp_nbr_sol_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
     _stale_nc(iface, &ipv6->src, l2src, l2src_len);
     memcpy(&nbr_adv_dst, &ipv6->src, sizeof(ipv6_addr_t));
 #endif
+    gnrc_ipv6_nc_t *nce = gnrc_ipv6_nc_get(iface, &ipv6->src);
+    if (nce != NULL) {
+        /* unset isRouter flag
+         * (https://tools.ietf.org/html/rfc4861#section-6.2.6) */
+        nce->flags &= ~GNRC_IPV6_NC_IS_ROUTER;
+    }
```

Comment:
Maybe I misunderstood the title and description of the PR, but shouldn't this go into `gnrc_ndp_rtr_sol_handle()` rather than into `gnrc_ndp_nbr_sol_handle()`?